# Big Data, Check Point 3: Credit Card Fraud Detection

In [1]:
from dask.distributed import Client, progress

client = Client()
client

<Client: 'tcp://127.0.0.1:41451' processes=2 threads=2, memory=13.65 GB>

In [2]:
import dask.dataframe as dd

In [4]:

# This loads the data into a Dask DataFrame
df = dd.read_csv('/content/drive/MyDrive/Thinkful/Big Data/creditcard.csv', 
                 dtype={'Time': 'float64'})

## 1. How many transactions do we have in total?

In [6]:
from dask import compute

compute(df.shape[0])

(284807,)

There are 284807 transactions in our dataset.

## 2. How many transactions are fraud and how many are not?

In [7]:
df.groupby("Class")["Time"].count().compute()

Class
0    284315
1       492
Name: Time, dtype: int64

We have 492 fraud transactions.

## 3. What is the maximum amount in fraud transactions?

In [8]:
df[df.Class == 1]["Amount"].max().compute()

2125.87

The maximum value of fraud tramsactions "Amount" 2125.87 units.